src/config.def

```diff
@@ -74,3 +75,10 @@ OPTION(bool, queue_profiling_use_timestamp_queries, false)
 OPTION(uint32_t, perfetto_trace_max_size, 1024u)
 OPTION(std::string, perfetto_trace_dest, "clvk.perfetto-trace")
 #endif // CLVK_PERFETTO_BACKEND_INPROCESS
+
+//
+// Test
+//
+#if CLVK_UNIT_TESTING_ENABLED
+OPTION(bool, force_descriptor_set_allocation_failure, false)
+#endif
```

Comment:
Good idea! I hadn't thought about that but a nice side effect of my PR is that we can use config options to drive test behaviours in many scenarios that would have required dedicated test functions.